langwatch/src/pages/api/thread/[id].ts

```diff
@@ -27,10 +28,11 @@ export default async function handler(
   }
 
   const threadId = req.query.id as string;
-
-  const traces = await getTracesByThreadId({
-    projectId: project?.id,
+  const protections = await getProtectionsForProject(prisma, { projectId: project?.id });
+  const traces = await getTracesGroupedByThreadId({
+    connConfig: { projectId: project?.id },
```

Comment:
This seems like a bit of a new pattern. Can we make this flat?

Reply:
We could here, but there are other places were we access via organization or in a test suit. Think this is nicer than 3-4 function signatures for each approach